web/src/drafts.ts

```diff
@@ -20,45 +21,71 @@ import * as timerender from "./timerender";
 import * as ui_util from "./ui_util";
 import * as util from "./util";
 
-export function set_count(count) {
+export function set_count(count: number): void {
     const $drafts_li = $(".top_left_drafts");
     ui_util.update_unread_count_in_dom($drafts_li, count);
 }
 
-export const draft_model = (function () {
-    const exports = {};
+type LocalStorageDraft = {
+    content: string;
+    updatedAt?: number;
```

Comment:
Why is this not marked as a required parameter? It looks like 1929cc5190387eab320966f067a78fe1cd728df7 has had that be stored when adding/editing drafts since the beginning of time, and 7aeae688e7da91ea185a1e0a17f00eee8b636cdb introduced auto-deleting old drafts back in 2018, so anything that might have been created in prototypes on chat.zulip.org before that was merged is long gone (and wouldn't be relevant anyway).

Reply:
Because `snapshot_message()` creates the draft without the timestamp, and the timestamp is added later. I can make separate types for these

Reply:
Actually maybe it makes sense to return the snapshot message with the timestamp already set? Because it's always added right after. I'll try that

Reply:
Yeah this looks nice.